src/models/sensor-polar-manager.ts

```diff
@@ -0,0 +1,243 @@
+  let str = num.toString(16);
+  return "0".repeat(2 - str.length) + str;
+};
+
+export class HeartRateSensorManager extends SensorManager {
+  device: BluetoothDevice | null;
+  server: BluetoothRemoteGATTServer | undefined;
+  _characteristic: BluetoothRemoteGATTCharacteristic | undefined;
+
+  private internalConfig: SensorConfig;
+  private hasData: boolean = false;
+  private stopRequested: boolean = false;
+
+  supportsDualCollection = false;
+
+  constructor() {
+    super();
+
+    this.internalConfig = {
+      collection:{ canControl:true, isCollecting:false },
+      columnListTimeStamp: new Date(),
+      columns:{
+        "100": {
+          id: "100",
+          setID: "100",
+          position: 0,
+          name: "Time",
+          units: "s",
+          liveValue: "NaN",
+          liveValueTimeStamp: new Date(),
+          valueCount: 0,
+          valuesTimeStamp: new Date()
+        },
+        "101": {
+          id: "101",
+          setID: "100",
+          position: 1,
+          name: "Heart Rate",
+          units: "bpm",
+          liveValue: "NaN",
+          liveValueTimeStamp: new Date(),
+          valueCount: 0,
+          valuesTimeStamp: new Date()
+        }
+       },
+      currentInterface: "Vernier Go Wireless Heart Rate",
+      currentState: "unknown",
+      os: { name: "Fake", version: "1.0.0"},
+      requestTimeStamp: new Date(),
+      server: { arch: "Fake", version: "1.0.0" },
+      sessionDesc: "Fake",
+      sessionID: "1234",
+      sets:{
+        "100": {
+          name: "Run 1",
+          colIDs: [100, 101]
+        }
+      }
+    };
+
+    this.supportsHeartbeat = true;
+  }
+
+  async connectToDevice(wirelessDevice: BluetoothDevice): Promise<boolean> {
+    this.device = wirelessDevice;
+
+    // Connect to device
+    this.server = await this.device?.gatt?.connect();
+
+    // Resend the sensorconfig so the UI udpates after the connection
+    this.sendSensorConfig(true);
+
+    // Get the Service
+    const service = await this.server?.getPrimaryService("heart_rate");
+
+    // Get and save data characteristic
+    this._characteristic = await service?.getCharacteristic('heart_rate_measurement');
+    if (!this._characteristic) { return false };
+
+    await this.startNotifications();;
+    this.handleHeartRateMeasurement(this._characteristic);
+
+    return true;
+  }
+
+  async disconnectFromDevice() {
+    if (this._characteristic) {
+      await this.stopNotifications();
+    }
+    this.device?.gatt?.disconnect();
+    // Resend the sensorconfig so the UI udpates after the disconnection
+    this.sendSensorConfig(true);
+  }
+
+  handleHeartRateMeasurement(heartRateMeasurement: BluetoothRemoteGATTCharacteristic) {
+    heartRateMeasurement.addEventListener('characteristicvaluechanged', (event: any) => {
+      const heartRateMeasurement = this.parseHeartRate(event.target?.value).heartRate;
+      this.printByteArray(event.target?.value);
```

Comment:
Do you still need this for debugging?

Reply:
Yes, it's helpful to keep there for debugging purposes.